services/QuillLMS/client/app/bundles/Staff/components/evidence/llmPromptDatasets/individualTrial/__tests__/viewGEvalModal.test.tsx

```diff
@@ -0,0 +1,60 @@
+import * as React from 'react';
+import { render, screen } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+
+import ViewGEvalModal from '../viewGEvalModal';
+import { GEvalInterface } from '../../../../../interfaces/evidenceInterfaces';
+
+describe('ViewGEvalModal', () => {
+  const mockGEval: GEvalInterface = {
+    id: 1,
+    metric: 'Accuracy',
+    version: 1,
+    task_introduction: 'This is a task introduction for testing.',
+    evaluation_criteria: 'These are the evaluation criteria.',
+    evaluation_steps: 'These are the evaluation steps.',
+    max_sure: 5,
+    misc: null,
+    selectable: true,
+    created_at: '2024-11-07T00:00:00Z',
+    updated_at: '2024-11-07T00:00:00Z',
+  };
+
+  const mockCloseModal = jest.fn();
+
+  const defaultProps = {
+    gEval: mockGEval,
+    closeModal: mockCloseModal
+  }
+
+  it('matches the snapshot with sample gEval data', () => {
+    const { asFragment } = render(
+      <ViewGEvalModal {...defaultProps} />
+    );
+    expect(asFragment()).toMatchSnapshot();
+  });
+
+  it('displays the correct G-Eval information', () => {
+    render(<ViewGEvalModal {...defaultProps} />);
+
+    expect(screen.getByText('Metric')).toBeInTheDocument();
+    expect(screen.getByText('Accuracy')).toBeInTheDocument();
+    expect(screen.getByText('Version')).toBeInTheDocument();
+    expect(screen.getByText('1')).toBeInTheDocument();
+    expect(screen.getByText('Task Introduction')).toBeInTheDocument();
+    expect(screen.getByText('This is a task introduction for testing.')).toBeInTheDocument();
+    expect(screen.getByText('Evaluation Criteria')).toBeInTheDocument();
+    expect(screen.getByText('These are the evaluation criteria.')).toBeInTheDocument();
+    expect(screen.getByText('Evaluation Steps')).toBeInTheDocument();
+    expect(screen.getByText('These are the evaluation steps.')).toBeInTheDocument();
```

Comment:
With this many values, my preferred approach of making a bunch of `const`s feels a bit heavy-handed.  That said, comparing the input and output, it may not be entirely obvious in this case whether the `expect(screen.getByText('1')).toBeInTheDocument();` refers to the `1` in `GEvalInterface.id` or the `1` in `GeEvalInterface.version`.  I'm pretty sure it's the latter based on context, but making that explicit by either using a `const` for at least that value, or changing the `version` or `id` value so that there aren't any duplicates in the input would be a bit clearer.